app/models/course/forum_settings.rb

```diff
@@ -27,18 +28,47 @@ def title=(title)
     @settings.title = title
   end
 
+  # Returns the title of topics in forums component
+  #
+  # @return [String] The custom or default title of topics in forums component
+  def topic_title
+    @settings.topic_title
+  end
+
+  # Sets the title of topics in forums component
+  #
+  # @param [String] title The new title
+  def topic_title=(title)
+    title = nil unless title.present?
+    @settings.topic_title = title
+  end
+
   # Returns the forum pagination count
   #
   # @return [Integer] The pagination count of forum
-  def pagination
-    @settings.pagination || 50
+  def forum_pagination
+    @settings.forum_pagination || 50
   end
 
   # Sets the forum pagination number
   #
   # @param [Integer] count The new pagination count
-  def pagination=(count)
-    @settings.pagination = count
+  def forum_pagination=(count)
+    @settings.forum_pagination = count
+  end
+
+  # Returns the topic pagination count
+  #
+  # @return [Integer] The pagination count of topic
+  def topic_pagination
```

Comment:
I think topic_pagination is for topic index page, and forum_pagination is for forum index page ?